src/components/Environments/Environment.service.ts

```diff
@@ -0,0 +1,120 @@
+} from '../app/details/triggerView/types'
+import { WebhookDetailsType, WebhookListResponse } from '../ciPipeline/Webhook/types'
+import { processWorkflow } from '../app/details/triggerView/workflow.service'
+import { WorkflowTrigger } from '../app/details/triggerView/config'
+import { Routes, URLS } from '../../config'
+import { get } from '../../services/api'
+import { ResponseType } from '../../services/service.types'
+
+export function getEnvWorkflowList(envId) {
+    const URL = `${Routes.ENV_WORKFLOW}/${envId}/${Routes.APP_WF}`
+    return get(URL)
+}
+
+export function getCIConfig(envID: number): Promise<ResponseType> {
+    const URL = `${Routes.ENV_WORKFLOW}/${envID}/${URLS.APP_CI_CONFIG}`
+    return get(URL)
+}
+
+export function getCDConfig(envID: number | string): Promise<ResponseType> {
+    const URL = `${Routes.ENV_WORKFLOW}/${envID}/${URLS.APP_CD_CONFIG}`
+    return get(URL)
+}
+
+export function getExternalCIList(envID: number | string): Promise<WebhookListResponse> {
+    return get(`${Routes.ENV_WORKFLOW}/${envID}/${URLS.APP_EXTERNAL_CI_CONFIG}`)
+}
+
+export const getWorkflowStatus = (envID: string) => {
+    return get(`${Routes.ENV_WORKFLOW}/${envID}/${Routes.WORKFLOW_STATUS}`)
+}
+
+export const getWorkflows = (envID): Promise<{ workflows: WorkflowType[]; filteredCIPipelines }> => {
+    const _workflows: WorkflowType[] = []
+    const _filteredCIPipelines = new Map()
+    return Promise.all([
+        getEnvWorkflowList(envID),
+        getCIConfig(envID),
+        getCDConfig(envID),
+        getExternalCIList(envID),
+    ]).then(([workflow, ciConfig, cdConfig, externalCIConfig]) => {
+        let _ciConfigMap = new Map<number, CiPipelineResult>()
+        for (const _ciConfig of ciConfig.result) {
+            _ciConfigMap.set(_ciConfig.appId, _ciConfig)
+        }
+        for (const workflowResult of workflow.result.workflows) {
+            const processWorkflowData = processWorkflow(
+                {
+                    ...workflowResult,
+                    workflows: [workflowResult],
+                } as WorkflowResult,
+                _ciConfigMap.get(workflowResult.appId),
+                cdConfig.result as CdPipelineResult,
+                externalCIConfig.result,
+                WorkflowTrigger,
+                WorkflowTrigger.workflow,
+                filterChildAndSiblingCD(envID),
+            )
+            _workflows.push(...processWorkflowData.workflows)
+            _filteredCIPipelines.set(workflowResult.appId, processWorkflowData.filteredCIPipelines)
+        }
+        return { workflows: _workflows, filteredCIPipelines: _filteredCIPipelines }
+    })
+}
+
+const filterChildAndSiblingCD = function (envID: number): (workflows: WorkflowType[]) => WorkflowType[] {
+    return (workflows: WorkflowType[]): WorkflowType[] => {
+        workflows.forEach((wf) => {
+            let nodes = new Map(wf.nodes.map((node) => [node.type + '-' + node.id, node] as [string, NodeAttr]))
```

Comment:
make it const

Reply:
Also remove commented

Reply:
It is re assigned below

Reply:
Looks like `wf.nodes` is getting re-assigned. Check & update.